src/main/java/com/metamx/common/StreamUtils.java

```diff
@@ -55,7 +55,9 @@ public static long copyToFileAndClose(InputStream is, File file) throws IOExcept
   {
     file.getParentFile().mkdirs();
     try (OutputStream os = new BufferedOutputStream(new FileOutputStream(file))) {
-      return ByteStreams.copy(is, os);
+      final long result =  ByteStreams.copy(is, os);
+      os.flush();
```

Comment:
we should add a comment that this is a workaround for JDK7 bug, that's been fixed in JDK8 http://hg.openjdk.java.net/jdk8/jdk8/jdk/rev/759aa847dcaf